src/monmove.cpp

```diff
@@ -536,6 +536,25 @@ static float get_stagger_adjust( const tripoint &source, const tripoint &destina
     return std::max( 0.01f, initial_dist - new_dist );
 }
 
+bool monster::die_if_drowning( const tripoint &at_pos, const int chance )
+{
+    if( g->m.has_flag( "LIQUID", at_pos ) && can_drown() && one_in( chance ) ) {
+        // if there's a vehicle here with a boardable part, the monster is on it
+        // and not drowning
+        const cata::optional<vpart_reference> bp = g->m.veh_at( at_pos ).part_with_feature( "BOARDABLE",
```

Comment:
```suggestion
        if( g->m.veh_at( at_pos ).part_with_feature( "BOARDABLE", false ) ) {
```
The variable `bp` is not needed.

Reply:
Fixed, thanks.